src/xtgeo/io/tsurf/_tsurf_reader.py

```diff
@@ -0,0 +1,799 @@
+            "The file is not a TSurf file. Expected '.ts' extension."
+        )
+
+
+@contextmanager
+def _get_text_stream(
+    file: FileLike, encoding: str
+) -> Generator[Union[TextIOWrapper, StringIO], None, None]:
+    """Context manager that yields a text stream.
+
+    Handles different input types:
+    - str: converts to Path and opens file with specified encoding
+    - Path objects: opens file with specified encoding
+    - BytesIO: wraps with TextIOWrapper with specified encoding
+    - StringIO: yields directly (already text-based)
+    """
+    if isinstance(file, (str, Path)):
+        file_path = Path(file) if isinstance(file, str) else file
+        with open(file_path, encoding=encoding) as stream:
+            yield stream
+    elif isinstance(file, BytesIO):
+        with TextIOWrapper(file, encoding=encoding) as text_wrapper:
+            yield text_wrapper
+    elif isinstance(file, StringIO):
+        yield file
+    else:
+        raise TypeError(f"Invalid type for 'file': {type(file)}")
+
+
+def read_tsurf(
+    file: FileLike,
+    encoding: str = "utf-8",
+) -> TSurfData:
+    """
+    Read a TSurf file and parse its triangulated surface data.
+    Note that only a subset of the TSurf format is currently supported,
+    more types of sections exist in the format specification.
+
+    Args:
+        file: Path to TSurf file (str or Path) or
+              a file-like object (BytesIO or StringIO).
+        encoding: Text encoding for reading the file
+
+    Returns:
+        TSurfData: Parsed surface data containing header, coordinate system,
+            vertices, and triangles
+
+    Raises:
+        FileNotFoundError: If file path doesn't exist or isn't a regular file
+        ValueError: If file extension isn't .ts or file format is invalid
+    """
+
+    # TSurf is a file format used in for example
+    # the GOCAD software. RMS can export triangulated surfaces in its
+    # structural model in the TSurf format.
+    # When unhandled keywords are present in the file, the processing is halted
+    # and an error message is issued. The user is expected to take action
+    # to fix the file.
+    # Documentation for the TSurf format is limited if you don't
+    # have access to the  GOCAD Developer's Guide, but
+    # here is one:
+    # - https://paulbourke.net/dataformats/gocad/gocad.pdf
+
+    # Determine error message context based on file type
+    filepath_errmsg: str
+    if isinstance(file, str):
+        filepath_errmsg = file
+    elif isinstance(file, Path):
+        filepath_errmsg = str(file)
+    elif isinstance(file, StringIO):
+        filepath_errmsg = "'input stream (StringIO)'"
+    elif isinstance(file, BytesIO):
+        filepath_errmsg = "'input stream (BytesIO)'"
+    else:
+        filepath_errmsg = "'input stream (unknown type)'"
+
+    if isinstance(file, (str, Path)):
+        file_path = Path(file) if isinstance(file, str) else file
+        _validate_file_path(file_path, filepath_errmsg)
```

Comment:
Some of this is duplicating functionality that already exists in the FileWrapper class:

https://github.com/equinor/xtgeo/blob/8b484c006d0e10c1ceb5e7d1b63b5d52964dae2f/src/xtgeo/io/_file.py#L98

I'd suggest taking the FileLike into the FileWrapper and using stuff already there.

https://github.com/equinor/xtgeo/blob/8b484c006d0e10c1ceb5e7d1b63b5d52964dae2f/src/xtgeo/io/_file.py#L595-L601

You have already added the 'is tsurf' file check there, for example, so you could remove `_is_tsurf_signature`, `_validate_file_path`, and perhaps some other things.